### PR TITLE
rgw: conform to the standard usage of string::find

### PR DESCRIPTION
**Pull request by Yan-waller**
Signed-off-by: Yan Jun yan.jun8@zte.com.cn

### CODE REVIEW

File: src/rgw/rgw_rest_s3.cc [hunk 1]
```diff
@@ -1258,14 +1258,14 @@ static void parse_params(const string& params_str, string& first,
...
       end = params_str.size();
 
     string param = params_str.substr(pos, end - pos);
 
-    int eqpos = param.find('=');
-    if (eqpos > 0) {
+    size_t eqpos = param.find('=');
+    if (eqpos != string::npos) {
       string param_name = rgw_trim_whitespace(param.substr(0, eqpos));
       string val = rgw_trim_quotes(param.substr(eqpos + 1));
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_s3.cc]: don't need val; avoid a possible copy

**Comment by mattbenjamin** [comment]
@Yan-waller one nit found (ideas on improving the clarity of this sort of code would be welcome:)

**Comment by Yan-waller** [comment]
@mattbenjamin thanks, I have fixed  it and kill some compiling warnings.

**Comment by Yan-waller** [comment]
retest this please

**Comment by theanalyst** [comment]
lgtm

File: src/rgw/rgw_rest_s3.cc [hunk 2]
```diff
@@ -1257,18 +1257,17 @@ static void parse_params(const string& params_str, string& first,
...
       end = params_str.size();
 
     string param = params_str.substr(pos, end - pos);
 
-    int eqpos = param.find('=');
-    if (eqpos > 0) {
-      string param_name = rgw_trim_whitespace(param.substr(0, eqpos));
-      string val = rgw_trim_quotes(param.substr(eqpos + 1));
-      params[param_name] = val;
+    size_t eqpos = param.find('=');
```
Comment (mattbenjamin) [inline src/rgw/rgw_rest_s3.cc]: can see why the original made steps explicit;  this is smaller but has the same effect, iiuc

**Pull request merged**